src/cpu/flexcpu/stld_forwarder.hh

```diff
@@ -0,0 +1,150 @@
+     *        have been registered through registerStore() beforehand!*
+     */
+    void commitStore();
+
+    /**
+     * Accessor for the name field. Added for use with tracing.
+     *
+     * @return The name of the StLdForwarder.
+     */
+    const std::string& name() const
+    { return _name; }
+
+    /**
+     * Use the store buffer data to populate the data dependencies through
+     * memory for a load, absent forwarding. Should be called only after the
+     * load has had its effective address calculated.
+     *
+     * @param inst_ptr A reference to the load instruction for which to
+     *                 populate dependencies.
+     */
+    void populateMemDependencies(
+        const std::shared_ptr<InflightInst>& inst_ptr);
+
+    /**
+     * Let the forwarder know about a memory barrier instruction. This should
+     * be done as soon as the barrier is decoded and identified. This allows
+     * searches through the store buffer to be shortened, since the barrier
+     * will impose stricter restrictions than older stores.
+     *
+     * @param inst_ptr The barrier to inform the forwarder about.
+     */
+    void registerMemBarrier(InflightInst* inst_ptr);
+
+    /**
+     * Let the forwarder know about a store instruction. This should be done as
+     * soon as the store is decoded and identified. This allows callbacks to be
+     * attached so that any subsequent loads can have data forwarded as soon as
+     * effective address and data is calculated for the store.
+     *
+     * @param inst_ptr The store to inform the forwarder about.
+     */
+    void registerStore(InflightInst* inst_ptr);
+
+    /**
+     * Request data for a load instruction. If a store has data for the load,
+     * then the data should be forwarded as soon as possilble. Otherwise, as
+     * early as is possible, the forwarder will forward the data through the
+     * callback, or inform that such is not possible by calling the callback
+     * with a nullptr packet.
+     *
+     * @param inst_ptr The load instruction for which data should be provided.
+     * @param req The (translated) request containing the address of the load.
+     * @param callback The callback through which the forwarded packet will be
+     *                 provided. The PacketPtr is only guaranteed to be valid
+     *                 for the duration of the call to the callback.
+     *                 Additionally, the callback may be made on the same call
+     *                 stack (before requestLoad() returns) if the latency for
+     *                 forwarding is configured to be zero. Packet is nullptr
+     *                 if forwarding is not possible.
+     */
+    void requestLoad(const std::shared_ptr<InflightInst>& inst_ptr,
+                     const RequestPtr& req,
+                     const std::function<void(PacketPtr)>& callback);
+};
+
+#endif // __CPU_FLEXCPU_LSQ_HH__
```

Comment:
Probably should have a newline at the end of the file. Most editors are configured this way...

Reply:
will change when I get in